python/lsst/ts/logging_and_reporting/web_app/services/jira_service.py

```diff
@@ -1,43 +1,74 @@
-import datetime
-import json
+import os
 import logging
 
-from pydantic import BaseModel
+from lsst.ts.logging_and_reporting.jira import JiraAdapter
 
 
 logger = logging.getLogger(__name__)
 
 
-class JiraTicket(BaseModel):
-    url: str
-    summary: str
-    updated: datetime.datetime
-    create: datetime.datetime
-    system: str
-    status: str
-    key: str
+INSTRUMENTS = {
+    "LATISS": "AuxTel",
+    "LSSTCam" : "Simonyi",
+}
+
+
+def filter_tickets_by_instrument(tickets, instrument):
+    """
+    Filters a list of JIRA tickets to include only those associated with a
+    specified instrument.
+
+    For each ticket, checks if any of the system names in the ticket's
+    'system' field match the instrument name or its corresponding value from
+    the INSTRUMENTS dictionary. If a match is found, adds a 'url' field to the
+    ticket containing a direct link to the JIRA ticket.
+
+    Args:
+        tickets (list of dict): List of JIRA ticket dictionaries,
+        each containing at least 'system' and 'key' fields.
+        instrument (str): The instrument name to filter tickets by.
+
+    Returns:
+        list of dict: Filtered list of tickets that match the instrument,
+        each with an added 'url' field.
+    """
```

Comment:
[We use Numpydoc](https://developer.lsst.io/python/style.html#documentation-strings-docstrings) standard for our python docstrings. Please take a look to https://developer.lsst.io/python/numpydoc.html#examples-of-method-and-function-docstrings.
```suggestion
    """Filters a list of JIRA tickets to include only those associated with a
    specified instrument.

    For each ticket, checks if any of the system names in the ticket's
    'system' field match the instrument name or its corresponding value from
    the INSTRUMENTS dictionary. If a match is found, adds a 'url' field to the
    ticket containing a direct link to the JIRA ticket.

    Parameters
    ----------
    tickets : `list[dict]`
        List of JIRA ticket dictionaries,
        each containing at least 'system' and 'key' fields.
    instrument : `str`
        The instrument name to filter tickets by.

    Returns
    ----------
    filtered_tickets : `list[dict]`
        Filtered list of tickets that match the instrument,
        each with an added 'url' field.
    """
```